packages/shared-ux/router/impl/routes.tsx

```diff
@@ -28,42 +29,47 @@ type RouterElementChildren = Array<
 
 export const Routes = ({
   legacySwitch = true,
+  enableExecutionContextTracking = false,
   children,
 }: {
   legacySwitch?: boolean;
+  enableExecutionContextTracking?: boolean;
   children: React.ReactNode;
 }) => {
   const match = useRouteMatch();
 
   return legacySwitch ? (
-    <Switch>{children}</Switch>
+    <SharedUXRoutesContext.Provider value={{ enableExecutionContextTracking }}>
+      <Switch>{children}</Switch>
+    </SharedUXRoutesContext.Provider>
   ) : (
-    <ReactRouterRoutes>
-      {Children.map(children as RouterElementChildren, (child) => {
-        if (React.isValidElement(child) && child.type === LegacyRoute) {
-          const path = replace(child?.props.path, match.url + '/', '');
-          const renderFunction =
-            typeof child?.props.children === 'function'
-              ? child?.props.children
-              : child?.props.render;
-
-          return (
-            <Route
-              path={path}
-              element={
-                <>
-                  <MatchPropagator />
-                  {(child?.props?.component && <child.props.component />) ||
-                    (renderFunction && renderFunction()) ||
-                    children}
-                </>
-              }
-            />
-          );
-        } else {
-          return child;
-        }
-      })}
-    </ReactRouterRoutes>
+    <SharedUXRoutesContext.Provider value={{ enableExecutionContextTracking }}>
+      <ReactRouterRoutes>
+        {Children.map(children as RouterElementChildren, (child) => {
+          if (React.isValidElement(child) && child.type === LegacyRoute) {
+            const path = replace(child?.props.path, match.url + '/', '');
+            const renderFunction =
+              typeof child?.props.children === 'function'
+                ? child?.props.children
+                : child?.props.render;
+            return (
+              <Route
+                path={path}
+                element={
+                  <>
+                    {enableExecutionContextTracking && <MatchPropagator />}
```

Comment:
do we need this here? Seems like this will also be called inside <Route/> component

Reply:
I don't think so, but I am not sure how to verify that. Would you mind checking it? If you and your team don't see a use case for it, I can remove it, but I am not sure why it was added in the first place.

Reply:
I double-checked. Agree that we should keep it here